enterprise/reporting/error_reporting.go

```diff
@@ -426,9 +436,33 @@ func (edr *ErrorDetailReporter) mainLoop(ctx context.Context, c types.SyncerConf
 				errorDetailReportsDeleteQueryTimer.Since(deleteReportsStart)
 				if err != nil {
 					edr.log.Errorf("[ Error Detail Reporting ]: Error deleting local reports from %s: %v", ErrorDetailReportsTable, err)
+				} else {
+					deletedRows += len(reports)
 				}
 				// vacuum error_reports_details table
-				edr.vacuum(ctx, dbHandle, c)
+				if deletedRows >= edr.config.GetInt("Reporting.errorReporting.vacuumThresholdDeletedRows", 100000) {
+					vacuumStart := time.Now()
+					if edr.vacuumFull.Load() {
+						if _, err := dbHandle.ExecContext(
+							ctx,
+							fmt.Sprintf("vacuum full analyze %s", pq.QuoteIdentifier(ErrorDetailReportsTable)),
+						); err != nil {
+							edr.log.Errorn("error vacuuming", logger.NewStringField("table", ErrorDetailReportsTable), obskit.Error(err))
+						} else {
+							deletedRows = 0
+						}
+					} else {
+						if _, err := dbHandle.ExecContext(
+							ctx,
+							fmt.Sprintf("vacuum analyze %s", pq.QuoteIdentifier(ErrorDetailReportsTable)),
+						); err != nil {
+							edr.log.Errorn("error vacuuming", logger.NewStringField("table", ErrorDetailReportsTable), obskit.Error(err))
+						} else {
+							deletedRows = 0
+						}
```

Comment:
This is all duplicated stuff, right? The only thing that changes is the query.

Why not simplify a bit:

```go
// vacuum error_reports_details table
if deletedRows >= edr.config.GetInt("Reporting.errorReporting.vacuumThresholdDeletedRows", 100000) {
	err := edr.vacuumTable(ctx, dbHandle, edr.vacuumFull.Load(), vacuumDuration)
	if err == nil {
		deletedRows = 0
	}
}

// ...

func (edr *ErrorDetailReporter) vacuumTable(ctx context.Context, dbHandle *sql.DB, full bool, stat stats.Timer) error {
	defer stat.RecordDuration()()
	query := "vacuum analyze %s"
	if full {
		query = "vacuum full analyze %s"
	}
	_, err := dbHandle.ExecContext(ctx, fmt.Sprintf(query, pq.QuoteIdentifier(ErrorDetailReportsTable)))
	if err != nil {
		edr.log.Errorn("error vacuuming", logger.NewStringField("table", ErrorDetailReportsTable), obskit.Error(err))
	}
	return err
}
```